armi/cases/suiteBuilder.py

```diff
@@ -167,11 +167,11 @@ def namingFunc(index, _case, _mods):  # pylint: disable=function-redefined
                     )
 
                 previousMods.append(type(mod))
-                mod(case.cs, case.bp, case.geom)
+                with case.cs._unlock():
+                    mod(case.cs, case.bp, case.geom)
```

Comment:
Just as a heads-up for when we take next steps on immutable settings, this is a good example of where modifying a CaseSettings object is perfectly reasonable. The `case.cs` here are in the process of being created, so either they haven't been locked in the first place, or `case.cs = case.cs.modified(blah)` would be reasonable here

Reply:
...I might even suggest leaving it as-is even, eating the warning until we figure out our next move

Reply:
Actually, you originally asked me to suppress those warnings. And this was the reason I encountered the `clone()` bug.

@jakehader pointed out that when he ran a case study it started out with dozens of `DeprecationWarning`s about immutable settings before the run even started. And you guys both agreed that was a valid place to change settings, as it wasn't mid-run at all.  It was something I had missed in my first pass with `DeprecationWarning`s, so I went off to fix it.

Then things exploded, and here we are.

We certainly need one more PR to make the settings immutable. But since it might be a big, breaking change for some people, I decided to give people a couple weeks of watching the `DeprecationWarning`s roll by before we make the change go live.

Reply:
yeah, i did 🤦 dont change it back, but just bear in mind when we actually tackle immutable settings

Reply:
Yeah, I am itching to work on making the settings fully immutable.  But trying to hold to my (completely arbitrary) two weeks notice situation.

Though... maybe I could get started on it this week, it will take a while for it to make it through review, I bet. Yeah, maybe I'll push that up the queue.